### ISSUE TITLE
ASSERT: [025]: zlib threw error: [-5] no space in buffer

### ISSUE THREAD

**Issue by ktosiek** [pgbackrest/pgbackrest#712]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: 2.12

2. PostgreSQL version: 11.2-1.pgdg18.04+1

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: Ubuntu 18.04.2 LTS on primary and standby, repository is hosted on an SMB share (I'm not sure which Windows version hosts it)

4. Did you install pgBackRest from source or from a package? package from http://apt.postgresql.org/

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

PostgreSQL log:
```
ASSERT: [025]: zlib threw error: [-5] no space in buffer
        STACK TRACE:
        common/compress/gzip/common:gzipError:85:(test build required for parameters)
            ... function(s) ommitted ...
        common/compress/gzip/compress:gzipCompressProcess:(trace log level required for parameters)
        common/io/filter/group:ioFilterGroupProcess:(trace log level required for parameters)
        common/io/read:ioReadInternal:(trace log level required for parameters)
        common/io/read:ioRead:(trace log level required for parameters)
        storage/storage:storageCopy:(debug log level required for parameters)
        command/archive/push/file:archivePushFile:(debug log level required for parameters)
        command/archive/push/push:cmdArchivePush:(debug log level required for parameters)
        main:main:(debug log level required for parameters)
2019-04-17 02:37:46.676 UTC [203798] LOG:  archive command failed with exit code 25
2019-04-17 02:37:46.676 UTC [203798] DETAIL:  The failed archive command was: env KRB5CCNAME=FILE:/tmp/krb5cc_113 pgbackrest --stanza=11-main archive-push pg_wal/0000000100003F3D0000002A
```

$ cat /etc/pgbackrest/pgbackrest.conf
```
[global]
repo-path=/mnt/idcst02/BackupPostresProd/
```
$ cat /etc/pgbackrest/conf.d/11-main.conf
```
[11-main]
pg1-path=/var/lib/postgresql/11/main
repo1-type=cifs
repo1-cipher-type=aes-256-cbc
repo1-retention-full=2
start-fast=y
stop-auto=y
#buffer-size=16MB

[11-main:archive-push]
compress-level=6
```
$ cat /etc/pgbackrest/conf.d/encryption.conf
```
[global]
repo1-cipher-pass=[REDACTED]
```

7. Describe the issue:
Archiving has stopped on a specific WAL file.
Setting compress-level to 9 for some time has helped - the problematic file went through, and after resetting compress-level to 6 archiving continues to work.

**Comment by guruguruguru** [comment]
We have the same problem right now on two machines. 
It seems setting the compress level to 9 helped here as well.

**Comment by dwsteele** [comment]
We have a solution for this -- and it seems to be related to #710.  More details tomorrow once we have completed testing.

**Comment by dwsteele** [comment]
This has been fixed at 2d73de1d360e and released in 2.13.

**Comment by dwsteele** [comment]
@guruguruguru If you would like to use a name other than `guruguruguru` in the release notes just let me know.

**Comment by dwsteele** [comment]
Closing since this should be fixed in 2.13.
